src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4679,7 +4679,7 @@ private bool IsConstructorAccessible(MethodSymbol constructor, ref HashSet<Diagn
 
                     if (forTargetTypedNew && method.IsDefaultValueTypeConstructor())
```

Comment:
>IsDefaultValueTypeConstructor [](start = 52, length = 29)

From our discussion and offline chat with Mads, the default constructor on structs should be allowed. As you pointed out, that allows scenarios with object initializers, like `new() { Property = value }`.

Reply:
Some further back-and-forth going on. This would mean `new()` could mean `0` when targeting `int` which isn't great. I'll get back to you.

Reply:
Couldn't we just handle it on the IDE side (e.g. where to use new(): no primitives, where type is evident, etc)? to me it doesn't seem to be actively harmful to be excluded from the feature.

Reply:
I was thinking about that, but Neal pointed out that what counts as a primitive isn't obvious (not a pre-existing concept in language spec). One notion of a primitive is types that can be constructed with a literal, but that would include tuples (which we do want to allow to target-type with `new(...)`).

We'll need to re-discuss in LDM.
Some options:
- allow
- disallow
- disallow on "primitives"
- disallow unless there is an initializer (but what about empty initializer, like `new() { }`?).

Reply:
Quick update: Decision from LDM today (2018-10-17) is to allow binding to default constructor of structs.